File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/ShapeFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    public abstract class ShapeFactory
    {
        private delegate ShapeBase ShapeMaker(string Parameters);
        private static readonly Dictionary<string, ShapeMaker> map = new Dictionary<string, ShapeMaker>()
        {
            {"Circle",    (Parameters) => new ShapeCircle(Parameters)    },
            {"Triangle",  (Parameters) => new ShapeTriangle(Parameters)  },
            {"Rectangle", (Parameters) => new ShapeRectangle(Parameters) },
        };
        public static readonly string[] Kinds = map.Keys.OrderBy(s => s).ToArray();
        public static ShapeBase make(string kind, string parameters)
        {
            return map[kind](parameters);
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO2/BridgeProject/src/ShapeFactory.java
import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;

public abstract class ShapeFactory
{
    private interface ShapeMaker { public ShapeBase make(String text); }
    private static HashMap<String,ShapeMaker> map=new HashMap<String,ShapeMaker>()
    {
        {
            put("Circle",    (text) -> new ShapeCircle(text));
            put("Triangle",  (text) -> new ShapeTriangle(text));
            put("Rectangle", (text) -> new ShapeRectangle(text));
        }
    };
    public static List<String> keys=map.keySet().stream().sorted().collect(Collectors.toList());
    public static ShapeBase make(String key,String text) { return map.get(key).make(text); }
}
